scripts/codegen/create_fortran_kernel_interface.py

```diff
@@ -45,6 +38,15 @@
 ''').lstrip('\n')
 
 
+def generate_edit_warning(fpath, comment_marker='!'):
+    return \
+        ('%s --------------[ DO NOT EDIT ]--------------\n' % comment_marker) + \
+        ('%s  THIS IS AN AUTOMATICALLY GENERATED FILE\n' % comment_marker) + \
+        ('%s  EDIT:\n' % comment_marker) + \
+        ('%s    %s\n' % (comment_marker, os.path.relpath(__file__, OCCA_DIR))) + \
+        (('%s    %s\n' %(comment_marker, os.path.relpath(fpath, OCCA_DIR))) if fpath else '') + \
+        ('%s ===========================================\n' % comment_marker)
+
```

Comment:
Using the f-strings looks like an improvement for readability, I didn't think of that. But, they're relatively recent (Python 3.6) so I don't know if it excludes people suffering on ancient RHELs and such? At least a version check at the top of the script would be nice I think?

Reply:
Python 3.6 is pretty old relative to the Python community, but having a version check is a good idea! The auto-generated code is also not expected to be run often so it should be fine to depend on >= 3.6

![image](https://user-images.githubusercontent.com/1812355/105970512-74077880-6057-11eb-8205-8078161fd644.png)

Reply:
Sure, just wanted to mention it since I'm forced to use ancient versions of things from time to time and it's always a hassle. Not an issue for my current use case though.